writers/writer.go

```diff
@@ -76,3 +92,70 @@ func (c *consumer) channelExists(channel string) bool {
 	_, found := c.channels[channel]
 	return found
 }
+
+func (c *consumer) subtopicExists(subtopic string) bool {
+	if _, ok := c.subtopics["*"]; ok {
+		return true
+	}
+
+	_, found := c.subtopics[subtopic]
+	return found
+}
+
+type filterConfig struct {
+	List []string `toml:"filter"`
+}
+
+type channelsConfig struct {
+	Channels filterConfig `toml:"channels"`
+}
+
+func LoadChannelsConfig(chanConfigPath string) map[string]bool {
+	data, err := ioutil.ReadFile(chanConfigPath)
+	if err != nil {
+		log.Print(fmt.Sprintf("Failed to open configuration file: %s", err))
+		log.Print(fmt.Sprintf("Default configuration is loaded"))
+		return getDefaultFilter()
+	}
+
+	var channelsCfg channelsConfig
+	if err := toml.Unmarshal(data, &channelsCfg); err != nil {
+		log.Print(fmt.Sprintf("Failed to unmarshal configuration file: %s", err))
+		log.Print(fmt.Sprintf("Default configuration is loaded"))
+		return getDefaultFilter()
+	}
+
+	channels := map[string]bool{}
+	for _, ch := range channelsCfg.Channels.List {
+		channels[ch] = true
+	}
+
+	return channels
+}
+
+type subtopicsConfig struct {
+	Subtopics filterConfig `toml:"subtopics"`
+}
+
+func LoadSubtopicsConfig(subtopicConfigPath string) map[string]bool {
+	data, err := ioutil.ReadFile(subtopicConfigPath)
+	if err != nil {
+		log.Print(fmt.Sprintf("Failed to open configuration file: %s", err))
+		log.Print(fmt.Sprintf("Default configuration is loaded"))
+		return getDefaultFilter()
+	}
+
+	var subtopicCfg subtopicsConfig
+	if err := toml.Unmarshal(data, &subtopicCfg); err != nil {
+		log.Print(fmt.Sprintf("Failed to unmarshal configuration file: %s", err))
+		log.Print(fmt.Sprintf("Default configuration is loaded"))
+		return getDefaultFilter()
+	}
+
+	subtopics := map[string]bool{}
+	for _, ch := range subtopicCfg.Subtopics.List {
+		subtopics[ch] = true
+	}
+
+	return subtopics
+}
```

Comment:
Can you replace Channels and Subtopics config and filter with one `Filter` (`FilterConfig`, `LoadFilter`, as well) that contains both channels and subtopics filters?

Reply:
Can I merge both files (channels.toml & subtopics.toml) into one file?

Reply:
@manuio What do you think about that?

Reply:
@jonathandreyer I think that 2 files it's easier to understand. I also prefer something retro-compatible.

Reply:
@jonathandreyer I don't mind if you merge files. I think that we can keep backward compatibility even if we merge files because we'll only be adding another TOML table.